metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1383,6 +1394,54 @@ private void cancelMaybeBalancePartitionLeaders() {
         queue.cancelDeferred(MAYBE_BALANCE_PARTITION_LEADERS);
     }
 
+    private static final String MAYBE_ELECT_UNCLEAN_LEADERS = "maybeElectUncleanLeaders";
+
+    private void maybeScheduleNextElectUncleanLeaders() {
+        if (uncleanScheduled != ImbalanceSchedule.SCHEDULED &&
+                replicationControl.areSomePartitionsLeaderless()) {
+            log.debug(
+                "Scheduling write event for {} because scheduled ({}), and areSomePartitionsLeaderless ({})",
+                MAYBE_ELECT_UNCLEAN_LEADERS,
+                uncleanScheduled,
+                replicationControl.areSomePartitionsLeaderless()
+            );
+
+            ControllerWriteEvent<Boolean> event = new ControllerWriteEvent<>(MAYBE_ELECT_UNCLEAN_LEADERS, () -> {
+                long startTimeNs = time.nanoseconds();
+                ControllerResult<Boolean> result = replicationControl.maybeBalancePartitionLeaders();
```

Comment:
I was testing it and wondered why it didn't work as expected. Turns out we invoke `maybeBalancePartitionLeaders` here, which should be `maybeElectUncleanLeaders`... :)